### ISSUE TITLE
[IDEA] Use ClickHouse feature `External Data for Query Processing` for data fetching

### ISSUE THREAD

**Issue by Felixoid** [lomik/graphite-clickhouse#59]
**UPD**: [This](https://clickhouse.tech/docs/en/engines/table-engines/special/external-data/) feature could be used to avoid problems with too long queries.

Besides, the maximum metrics per query should be checked

---
Hello. I would like to implement a feature to mitigate an error `DB::Exception: Syntax error: failed at position 262125: ..... Max query size exceeded`

The idea is to get the current user setting from ClickHouse. Then split the query https://github.com/lomik/graphite-clickhouse/blob/master/render/handler.go#L162 to multiple and send then in parallel. The setting could be refreshed as well as rollup config in background once per minute.

What do you think?

**Comment by lomik** [comment]
I would suggest to try another way first. You can send series list as attached file as described here https://clickhouse.yandex/docs/en/single/#external-data-for-query-processing . I want to try it for a long time but there is not enough time

Usually it is enough to increase this parameter through the settings 
```
[clickhouse]
url = "http://graphite:qwerty@localhost:8123/?max_query_size=2097152&max_ast_elements=1000000&readonly=2&log_queries=1"
```

**Comment by Felixoid** [comment]
AFAIK, it's forbidden to change settings for users with `readonly=1` in CH. So, if anybody does have real readonly user, it won't work

**Comment by Felixoid** [comment]
Thank you for the suggestion of using external tables, I'll try to implement it.

**Comment by Hipska** [comment]
I think this issue could also be solved when #22 get's resolved.

**Comment by Felixoid** [comment]
PoC is ready in the PR, you are very welcome to try it out
